automation/Commands/command_executor.py

```
@@ -1,66 +1,74 @@
+            visit_id=command.visit_id, driver=webdriver,
+            manager_params=manager_params, suffix=command.suffix)
 
-    if command[0] == 'RECURSIVE_DUMP_PAGE_SOURCE':
+    if type(command) is Recursive_Dump_Page_Source_Command:
         browser_commands.recursive_dump_page_source(
-            visit_id=command[2], driver=webdriver,
-            manager_params=manager_params, suffix=command[1])
+            visit_id=command.visit_id, driver=webdriver,
+            manager_params=manager_params, suffix=command.suffix)
 
-    if command[0] == 'SAVE_SCREENSHOT':
+    if type(command) is Save_Screenshot_Command:
         browser_commands.save_screenshot(
-            visit_id=command[2], crawl_id=browser_params['crawl_id'],
-            driver=webdriver, manager_params=manager_params, suffix=command[1])
+            visit_id=command.visit_id, crawl_id=command.crawl_id,
+            driver=webdriver, manager_params=manager_params,
+            suffix=command.suffix)
 
-    if command[0] == 'SCREENSHOT_FULL_PAGE':
+    if type(command) is Screenshot_Full_Page_Command:
         browser_commands.screenshot_full_page(
-            visit_id=command[2], crawl_id=browser_params['crawl_id'],
-            driver=webdriver, manager_params=manager_params, suffix=command[1])
+            visit_id=command.visit_id, crawl_id=command.crawl_id,
+            driver=webdriver, manager_params=manager_params,
+            suffix=command.suffix)
 
-    if command[0] == 'RUN_CUSTOM_FUNCTION':
+    if type(command) is Run_Custom_Function_Command:
         arg_dict = {"command": command,
                     "driver": webdriver,
                     "browser_settings": browser_settings,
                     "browser_params": browser_params,
                     "manager_params": manager_params,
                     "extension_socket": extension_socket}
-        command[1](*command[2], **arg_dict)
+        command.function_handle(*command.func_args, **arg_dict)
```

Could you change all of these to `elif`s and throw an exception when an invalid command gets passed?
This way invalid commands can be caught in development instead of just being ignored